### DISCUSSION TITLE
Mis2Mean error

### DISCUSSION THREAD

**Discussion by lilarou** [mtex-toolbox/mtex#1846] [Ask Anything]
Hi MTEX community,

I have EBSD data from which I'd like to get the GROD and GOS values. 
I define grains and remove all boundary grains (_grains_int_) because I want grain statistics : 

```matlab
ebsd = EBSD.load(fname,CS,'interface','crc',...
  'convertSpatial2EulerReferenceFrame');
ebsdTi = ebsd('Ti-Hex');

[grains,ebsdTi.grainId]= calcGrains(ebsdTi,'angle',2*degree);
grains = smooth(grains,5);
grains_int=grains(~grains.isBoundary);

orientation = grains_int.meanOrientation;
```

The grains list contains 5137 grains and the grains_int list contains 4844 grains.

When I call the function `mis2mean = calcGROD(ebsdTi, **grains**);` it works but the function `mis2mean = calcGROD(ebsdTi, **grains_int**);` returns the following error : _Index exceeds the number of array elements (4844)._
I don't understand why I can't get the mis2mean value for the grains_int.
Do you know how I could get it ?

Thank you

**Comment by kilir** [comment]
Hi,
this error is expected since the indices do not match the data.
How about about this?
```
grains.mis2mean = calcGROD(ebsdTi, grains);
grains(~grains.isBoundary)
% or alternatively
[grains,ebsd.grainId, ebsd.mis2mean] = ebsd.calcGrains
```
Cheers.
Rüdiger

**Reply by lilarou** [reply]
Hi Rudiger,

Your suggestion do not match because 'mis2mean' is not defined for the 'grain2d' class

**Reply by kilir** [reply]
Hi,
yes you are right.
it should have been `ebsd.mis2mean = calcGROD(ebsd, grains);`
Cheers,
Rüdiger

**Comment by ralfHielscher** [comment]
You can do
```matlab
mis2mean = calcGROD(ebsdTi(grains_int), grains_int);
```

**Reply by lilarou** [reply]
Unfortunately, I have the same error (_Index exceeds the number of array
elements (4844)._) even by specifying 'ebsdTi(grains_Ti)'
